src/main/java/org/alliancegenome/curation_api/dao/base/BaseSQLDAO.java

```diff
@@ -147,18 +163,44 @@ public E merge(E entity) {
 	public E remove(String id) {
 		log.debug("SqlDAO: remove: " + id);
 		E entity = find(id);
-		entityManager.remove(entity);
+
+		try {
+			entityManager.remove(entity);
+			entityManager.flush();
+		} catch (PersistenceException e) {
+			handlePersistenceException(entity, e);
+		}
 		return entity;
 	}
 
 	@Transactional
 	public E remove(Long id) {
 		log.debug("SqlDAO: remove: " + id);
 		E entity = find(id);
-		entityManager.remove(entity);
+
+		try {
+			entityManager.remove(entity);
+			entityManager.flush();
+		} catch (PersistenceException e) {
+			handlePersistenceException(entity, e);
+		}
 		return entity;
 	}
 	
+	private void handlePersistenceException(E entity, Exception e) {
+		ObjectResponse<E> response = new ObjectResponse<E>(entity);
+		Throwable rootCause = e.getCause();
```

Comment:
@markquintontulloch this should get changed to getRootCause() plus this code shouldn't be here in the DAO.

Reply:
@oblodgett - getRootCause() doesn't appear to exist.  I agree about the code not being in the DAO, but moving it to the controller seemed like a big job outside the scope of this ticket as it affects everything that calls the DAO remove method.